be/src/vec/exec/format/csv/csv_reader.cpp

```diff
@@ -462,8 +462,22 @@ void CsvReader::_split_line_for_single_char_delimiter(const Slice& line) {
         size_t cur_pos = 0;
         size_t start_field = 0;
         const size_t size = line.size;
+        bool quoted = false;
+
         for (; cur_pos < size; ++cur_pos) {
-            if (*(value + cur_pos) == _value_separator[0]) {
+            if (_trim_double_quotes && *(value + cur_pos) == '\"') {
+                if (cur_pos == start_field) {
+                    // start_field is ", mark quoted true
+                    quoted = true;
+                } else if (*(value + cur_pos - 1) != '\\' && quoted) {
+                    // cur_pos is ", mark quoted finished
+                    quoted = !quoted;
+                }
+                continue;
+            } else if (_trim_double_quotes && quoted) {
```

Comment:
Better leave `_split_line_for_single_char_delimiter` fast enough to not handle this complex situation.
Maybe we should be a new parameter to specify the `escape` explicitly.

Reply:
> 



> Better leave `_split_line_for_single_char_delimiter` fast enough to not handle this complex situation. Maybe we should be a new parameter to specify the `escape` explicitly.

done